src/cpp/rtps/builtin/liveliness/WLP.cpp

```diff
@@ -138,18 +139,48 @@ WLP::~WLP()
     {
         mp_participant->deleteUserEndpoint(mp_builtinReaderSecure);
         mp_participant->deleteUserEndpoint(mp_builtinWriterSecure);
-        delete this->mp_builtinReaderSecureHistory;
-        delete this->mp_builtinWriterSecureHistory;
+
+        if (mp_builtinReaderSecureHistory)
+        {
+            PoolConfig sreader_pool_cfg = PoolConfig::from_history_attributes(mp_builtinReaderSecureHistory->m_att);
+            delete mp_builtinReaderSecureHistory;
+            secure_payload_pool_->release_history(sreader_pool_cfg, true);
+        }
+
+        if (mp_builtinWriterSecureHistory)
+        {
+            PoolConfig swriter_pool_cfg = PoolConfig::from_history_attributes(mp_builtinWriterSecureHistory->m_att);
+            delete mp_builtinWriterSecureHistory;
+            secure_payload_pool_->release_history(swriter_pool_cfg, false);
+        }
+
+        TopicPayloadPoolRegistry::release(secure_payload_pool_);
```

Comment:
I know that right now every time we construct a new WLP, the `payload_pool_` and `secure_payload_pool_` are retrieved from the registry. But as an extra security for the future, it would be good if we checked if the pointer is set to something before releasing it.

Reply:
I don't think this is necessary. `TopicPayloadPoolRegistry::release` already handles the case of receiving an unset pointer. Even more, if there is a history created, it should be assigned. Otherwise, a segfault will happen, as we are using it previously to release the histories.

Reply:
My concerns were more about having a WLP object not initialized, this having no history either. In any case, I can see what you mean about the registry already doing the check, so that's good enough to me.